core/app/models/concerns/spree/user_address_book.rb

```diff
@@ -56,17 +56,19 @@ def default_address_attributes=(attributes)
 
       def ship_address=(address)
         be_default = Spree::Config.automatic_default_address
-        save_in_address_book(address.attributes, be_default)
+        save_in_address_book(address.attributes, be_default) if address
       end
 
       def persist_order_address(order)
```

Comment:
@allisonlarson would you mind also adding some quick docs here while we're in the area?